build-system/eslint-rules/no-non-string-log-args.js

```diff
@@ -0,0 +1,130 @@
+  {name: 'assertString', variadic: false, startPos: 1},
+  {name: 'assertNumber', variadic: false, startPos: 1},
+  {name: 'assertBoolean', variadic: false, startPos: 1},
+  {name: 'assertEnumValue', variadic: false, startPos: 2},
+  {name: 'assertElement', variadic: false, startPos: 1},
+  {name: 'createExpectedError', variadic: true, startPos: 0},
+  {name: 'fine', variadic: true, startPos: 1},
+  {name: 'info', variadic: true, startPos: 1},
+  {name: 'warn', variadic: true, startPos: 1},
+  {name: 'error', variadic: true, startPos: 1},
+  {name: 'expectedError', variadic: true, startPos: 1},
+  {name: 'createError', variadic: true, startPos: 0},
+];
+
+/**
+ * @param {!Node} node
+ * @return {boolean}
+ */
+function isMessageString(node) {
+  if (node.type === 'Literal') {
+    return typeof node.value === 'string';
+  }
+  // Allow for string concatenation operations.
+  if (node.type === 'BinaryExpression' && node.operator === '+') {
+    return isMessageString(node.left) && isMessageString(node.right);
+  }
+  return false;
+}
+
+/**
+ * @param {string} name
+ * @return {!LogMethodMetadataDef}
+ */
+function getMetadata(name) {
+  return transformableMethods.find(cur => cur.name === name);
+}
+
+const expressions = transformableMethods.map(method => {
+  return `CallExpression[callee.property.name=${method.name}]`;
+}).join(',');
+
+
+module.exports = function(context) {
+  return {
+    [expressions]: function(node) {
+      // Make sure that callee is a CallExpression as well.
+      // dev().assert() // enforce rule
+      // dev.assert() // ignore
+      if (!node.callee.object ||
+          node.callee.object.type !== 'CallExpression') {
+        return;
+      }
+
+      // Make sure that the CallExpression is one of dev() or user().
+      if(!['dev', 'user'].includes(node.callee.object.callee.name)) {
+        return;
+      }
+
+      const methodInvokedName = node.callee.property.name;
+      // Find the position of the argument we care about.
+      const metadata = getMetadata(methodInvokedName);
+
+      // If there's no metadata, this is most likely a test file running
+      // private methods on log.
+      if (!metadata) {
+        return;
+      }
+
+      const argToEval = node.arguments[metadata.startPos];
+
+      if (!argToEval) {
+        context.report({
+          node: node,
+       	  message: `No argument passed into expected message parameter ` +
+            `number ${metadata.startPos}`,
+        });
+        return;
+      }
+
+      let errMsg = [
+        `Must use a literal string for argument[${metadata.startPos}]`,
+        `on ${metadata.name} call.`
+      ].join(' ');
+
+      if (metadata.variadic) {
+        errMsg += 'If you want to pass data to the string, use `%s` ';
+        errMsg += 'placeholders and pass additional arguments';
+      }
+
+      if (!isMessageString(argToEval)) {
+        context.report({
+          node: argToEval,
+       	  message: errMsg,
```

Comment:
You have some weird whitespacing...

Reply:
@jridgewell i think its cause i keep on copy pasting from ast explorer =/. aplogies